plumbing/transport/common.go

```diff
@@ -228,10 +229,10 @@ func parseSCPLike(endpoint string) (Endpoint, bool) {
 	}
 
 	m := scpLikeUrlRegExp.FindStringSubmatch(endpoint)
-	return &scpEndpoint{
-		user: m[1],
-		host: m[2],
-		path: m[3],
+	return &SCPEndpoint{
+		Username: m[1],
+		Hostname: m[2],
+		Pathname: m[3],
```

Comment:
Happy to rename it if you can think of something useful to name it. Alternatively I could convert it into a type alias of a string instead of a struct.